components/automate-ui/src/app/modules/infra-proxy/cookbook-details/cookbook-details.component.ts

```diff
@@ -133,6 +166,41 @@ export class CookbookDetailsComponent implements OnInit, OnDestroy {
     }));
   }
 
+  listClickEvent(event) {
+    if (event.target.classList.contains('extend-list')) {
+      event.target.classList.remove('extend-list');
+      event.target.parentNode.querySelector('ul').classList.remove('show');
+    } else {
+      event.target.classList.add('extend-list');
+      event.target.parentNode.querySelector('ul').classList.add('show');
+    }
+  }
+
+  subListClickEvent(event, currentContent: SubMenu) {
+    this.contentLoading = true;
+    this.getContent(currentContent);
+    document.querySelector('li.active-sub-list').classList.remove('active-sub-list');
+    event.target.classList.add('active-sub-list');
+  }
+
+  getContent(activeContent: SubMenu) {
+    this.activeContentName = activeContent.name;
+    this.contentUrl = encodeURIComponent(activeContent.url);
+    this.http.get(
+      `${env.infra_proxy_url}/servers/${this.serverId}/orgs/${this.orgId}/cookbooks/${this.cookbookName}/${this.currentVersion}/file-content?url=${this.contentUrl}`)
```

Comment:
This goes against two well-established conventions:
1. All backend requests should be done via an ngrx dispatch.
2. Avoid nested subscriptions.

Looks like there is another such occurrence in this same file. Please fix this as well:
https://github.com/chef/automate/blob/aabcc5c2c90f3e996b203f73a60832c109e5b4da/components/automate-ui/src/app/modules/infra-proxy/cookbook-details/cookbook-details.component.ts#L112-L115

Reply:
@msorens as we are reading the file content and showing that on UI.
1. for your first point is that be OK to manage whole file content (file size can be vary ) which we getting with ngrx store module.
2. for now nested subscription handled by explicitly completing the subscription using `first` as suggested here https://github.com/chef/automate/pull/3356#discussion_r410454267

Reply:
1. Good question; I am not really sure so let's keep what you have for that.
2. Yes, you are correct, using `first` is sufficient here, too (as I said in the comment 😁  you linked to).